docs/api-communication-spec.md

```diff
@@ -0,0 +1,686 @@
+    - Message
+      - Stop
+      - Congestion
+      - Error
+        - Limit reached (e.g. number of concurrent capture requests exceeded, )
+        - Invalid Request (e.g. malformed filter, device not found, etc.)
+        - Runtime (any unexpected error occurring on the agent during capture. Contains as much information as possible and forward before terminating)
+    - captured pcap data packet(s) - consider batching multiple packets in one Capture Response
+### pcap-api -> pcap-cli
+
+- Status
+- captured pcap data packet
+- Message
+    - Stop confirmation
+    - Capture status
+        - Agent is gone
+        - All agents stopped
+        - Congestion
+    - Errors
+        - Limit reached (e.g. number of concurrent capture requests exceeded, )
+        - Unauthorized
+        - Invalid Request (e.g. malformed filter, Target not found, device not found, etc.)
+
+## Data Structures
+
+The following section defines the data structures, their use and fields' purpose in detail.
+
+### Capture Request
+
+A capture request contains the information needed to target a set of resources, from which to capture network traffic.
+
+- **For CF**, a resource is a single application instance.
+  - The capture request can target all instances at once by omitting specific application instance IDs
+- **For BOSH**, a resource is a single BOSH VM.
+  - The capture request can target individual VMs or all VMs in one or multiple instance groups of a BOSH deployment.
+
+The capture request can contain a start or stop request. The start request can either be for CF or BOSH respectively:
+
+* `start`: one of [CF App Start Capture Request](#cf-app-start-capture-request), [BOSH Start Capture Request](#bosh-start-capture-request)
+* `stop`: a [Stop Capture Request](#stop-capture-request)
+
+#### Common Fields for `pcap-agent`
+
+The following attributes are part of all capture requests and define the details required for the `pcap-agent`.
+
+| Parameter      | Type    | Required? | Default | Description                                                                                                                                               |
+|----------------|---------|-----------|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `device`       | string  | yes       | `eth0`   | The device on the app container from which to capture the traffic                                                                                         |
+| `filter`       | string  | no        |         | BPF Filter expression to use with tcpdump command to capture traffic.                                                                                     |
+| `snaplen`      | integer | no        | 65535   | Limit the amount of data captured for each packet, see [SnapLen](https://wiki.wireshark.org/SnapLen)                                                      |
```

Comment:
Either format the tables or don't. But I don't like this, applies to the tables below as well.

Reply:
Found the format button in IJ